src/visit/mod.rs

```diff
@@ -534,44 +534,40 @@ pub trait VisitMap<N> {
     /// Mark `a` as visited.
     ///
     /// Return **true** if this is the first visit, false otherwise.
+    ///
+    /// **Panics** if `a` is out of bounds.
     fn visit(&mut self, a: N) -> bool;
 
     /// Return whether `a` has been visited before.
+    ///
+    /// **Panics** if `a` is out of bounds.
```

Comment:
I'm sorry to be so detail-oriented, but it's good for the API if we are. With this formulation, we promise to panic if `a` is out of bounds - all implementations don't actually do that (HashSet based ones don't). So the doc would need to be updated to reflect that (I'd say *may* panic, then).

Reply:
No worries, I fixed the formulation.
```
    /// May **panic** if `a` is out of bounds.
```